### DISCUSSION TITLE
GitHub OAuth Client in Reader API

### DISCUSSION THREAD

**Discussion by tyteen4a03** [Thinkmill/keystatic#884] [Q&A] [answered]
Currently, the GitHub reader does not support OAuth-based authentication. It also ignores Keystatic configs if given one.

I can see the value of creating reader APIs for potentially a different branch than what's allowed in Keystatic, but I think the authentication config should be the same as the rest of Keystatic. 

----

EDIT: It looks like I might be using the script incorrectly - it looks like the GitHub Reader simply has no support for being used outside of an live Next app; is this true?

**Comment by emmatown** [comment, marked as answer]
The GitHub reader just accepts a GitHub token, you can get that however you'd like to. For example, you could use the token that Keystatic uses that's stored in a cookie named `keystatic-gh-access-token`, a personal access token or anything else. It's not at all restricted to Next.

**Reply by tyteen4a03** [reply]
Yes, but what I wanted to get at is when I pass in the `keystaticConfig`, it should fetch the token for me (unless it's impossible? I haven't studied too much into how non-PAT access token fetching works)

**Reply by emmatown** [reply]
The GitHub reader can't do that because it doesn't have a place to get a token from, it doesn't know it's in a request and where to get the cookies from or etc.

**Reply by tyteen4a03** [reply]
Ah I see, it looks like GitHub doesn't support implicit flows so that's why it doesn't work that way. Thanks for explaining!

**Comment by tyteen4a03** [comment]
Based on the discussion here I've created a PR to improve the docs: https://github.com/Thinkmill/keystatic/pull/886
